client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -162,25 +165,50 @@ const Checkout = React.createClass( {
 	},
 
 	getCheckoutCompleteRedirectPath: function() {
-		var renewalItem,
+		var product,
+			purchasedProducts,
+			renewalItem,
+			receipt = this.props.transaction.step.data,
 			receiptId = ':receiptId';
 
 		this.props.clearPurchases();
 
 		if ( cartItems.hasRenewalItem( this.props.cart ) ) {
 			renewalItem = cartItems.getRenewalItems( this.props.cart )[ 0 ];
+			// group all purchases into an array
+			purchasedProducts = reduce( receipt && receipt.purchases || {}, function( result, value ) {
+				return result.concat( value );
+			}, [] );
+			// and take the first product which matches the product id of the renewalItem
+			product = find( purchasedProducts, function( item ) {
```

Comment:
Not a necessary change, but something you mind find interesting: _.find allows you to use an object as a predicate, like:

```js
purchase = find( purchases, { product_id: renewalItem.product_id } );
```